### ISSUE TITLE
SSH fails with `${HOME}` permission `drwxr-s---`

### ISSUE THREAD

**Issue by derek-burdick** [tailscale/tailscale#8118]
### What is the issue?

When I ssh to server over tailnet I receive Connection to 100.106.100.25 closed immediately.

I expect authentication to succeed like it does for normal sshd.

Error is the following:
```
May 10 20:36:04 qqqqq tailscaled[10697]: ssh-session(sess-20230511T033604-7bc50db3b2): access granted to user@user.com as ssh-user "ldaptestuser"
May 10 20:36:04 qqqqq tailscaled[10697]: ssh-session(sess-20230511T033604-7bc50db3b2): starting pty command: [/usr/sbin/tailscaled be-child ssh --uid=4007 --gid=2564 --groups=2564 --local-user=ldaptestuser --remote-user=user@user.com>
May 10 20:36:04 qqqqq tailscaled[10697]: ssh-session(sess-20230511T033604-7bc50db3b2): start failed: fork/exec /usr/sbin/tailscaled: permission denied
```

### Steps to reproduce

System is configured with Redhat authselect, configured against LDAP.

### Are there any recent changes that introduced the issue?

I am using latest source code which solves user not found issue from #8098. Next issue is permission related.
Git commit cb2fd5be922eb048c2f4f85a529082b1913a9dbd

Built using command:
```bash
TS_USE_TOOLCHAIN=1 ./build_dist.sh tailscale.com/cmd/tailscaled
```

### OS

Linux

### OS version

Linux 5.14.0-162.6.1.el9_1.x86_64 #1 SMP PREEMPT_DYNAMIC Fri Sep 30 07:36:03 EDT 2022 x86_64 x86_64 x86_64 GNU/Linux

### Tailscale version

1.41.55

### Other software

Redhat SSSD and Authselect
SELINUX is disabled

### Bug report

BUG-6bbc532ccfa255db2925d630b88107fec4525ee345569ffe7197be1f3431325c-20230511033358Z-9019d864fc06f0be

**Comment by bradfitz** [comment]
Kinda smells like SELinux, even though you say it's disabled. Anything similar security-wise special on there?

**Comment by derek-burdick** [comment]
That was my first thought as well. SELINUX is disabled.
```
[qqqq]$ sudo sestatus
SELinux status:                 disabled
```

Tailscale SSH is working for locally defined users, so it is something related to the dynamically defined ones.

**Comment by derek-burdick** [comment]
I've confirmed it is lack of permissions on users home directory.

This corrects the issue. I'm going to do some additional tests and see what needs to be done.
```diff
diff --git a/ssh/tailssh/incubator.go b/ssh/tailssh/incubator.go
index 1a277d63..e347a579 100644
--- a/ssh/tailssh/incubator.go
+++ b/ssh/tailssh/incubator.go
@@ -429,6 +429,7 @@ func (ss *sshSession) launchProcess() error {
        } else {
                return err
        }
+       cmd.Dir = "/"
        cmd.Env = envForUser(ss.conn.localUser)
        for _, kv := range ss.Environ() {
                if acceptEnvPair(kv) {
```

Here are the directory permissions in /home:
```
drwxr-s--- 6 localuser localuser 9 May 11 08:49 localuser
```

**Comment by bradfitz** [comment]
@derek-burdick, interesting! Let us know what more you find.

**Comment by derek-burdick** [comment]
cmd.Dir is the directory used by the spawned 'tailscaled ssh' instance. It has no impact on the directory of the shell my user gets. I would propose that we either set cmd.Dir = '/' in all cases, maybe set it based on OS?

Or alternatively, and I have verified it works, we just leave it empty, which should use the current processes working directory.

**Comment by bradfitz** [comment]
I'm a little skeptical, but maybe? I'll have to research more before accepting your #8120. Did you also try non-interactive sessions? (ssh you@tailscale some-command)

BTW, please update your #8120's commit message to match our conventional format. See `git log` for examples. It's effectively Go's style: https://golang.org/wiki/CommitMessage

**Comment by derek-burdick** [comment]
Good call on non-interactive. It indeed is in the wrong directory. I'm going to move the homeDir lookup code in to the incubator/ssh process, instead of the parent process. I've tested a quick hack and i'm working on the fix now. Should have something ready shortly.

Do you know of a way to test the runningAsRoot && ia.loginCmdPath != "" code path in incubator.go:beIncubator function? Based on the comments of unix.Exec, i'm planning on just setting environment variable PWD, which is implied that sets the running directory.

**Comment by derek-burdick** [comment]
@bradfitz  I have tested with local users on centos 7 and redhat 9. I have also tested with sssd dynamic users on redhat 9 and dynamic users on centos7 using Azure SSH for linux.

Tests for non-interactive and interactive commands all work.

Also verified steps from https://github.com/tailscale/tailscale/issues/4908#issuecomment-1382941850 resolve issues with SELINUX and work with local and dynamic users, interactive and non-interactive.

Let me know your thoughts.
